CNFT1-331: Create operator dropdown component

joeaxcdw opened the pull request:
## Description

Implement the base operator dropdown to use for first name, last name, and applicable address fields in patient search.

![image](https://github.com/user-attachments/assets/36425f36-0ea5-425f-8088-e74659a5f2ab)

## Tickets

* [Jira Ticket](https://cdc-nbs.atlassian.net/browse/CNFT1-3317)

## Checklist before requesting a review
- [X] PR focuses on a single story
- [X] Code has been fully tested to meet acceptance criteria
- [X] PR is reasonably small and reviewable (Generally less than 10 files and 500 changed lines)
- [X] All new functions/classes/components reasonably small
- [X] Functions/classes/components focused on one responsibility
- [X] Code easy to understand and modify (clarity over concise/clever)
- [X] PRs containing TypeScript follow the [Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)
- [X] PR does not contain hardcoded values (Uses constants)
- [X] All code is covered by unit or feature tests

joeaxcdw commented on apps/modernization-ui/src/options/operator/operators.ts:
```diff
@@ -0,0 +1,26 @@
+import { Selectable, asSelectable, findByValue } from 'options';
+
+const STARTS_WITH_OPERATOR = asSelectable('S', 'Starts with');
```
@stevegsa please let me know what the actual values will be for these operators so I can fix it here.

adamloup-enquizit reviewed (commented):
This component is more than a dropdown of operators, it's encapsulating a specific use case searching using text in a specific way.  This is going to be a repeatable use case so the component should produce an object representing the intention of searching with this criteria.

At least something like 

```json
 {
 "value": "some value",
 "operator": { "name": "Starts with", "value" : "SW"}
}
```  

Even better would be something way more expressive 

```json
{
 "startsWith" :  "some-value"
}
```

---

Since this is going to represent a common usage pattern it should be in the `design-system` folder.

joeaxcdw commented:
Recent updates:

- Added `OperatorInput` control that combines the operator select and input
- Updated operation names and object types to match:
```
"city": {
  "startsWith": "some city"
 }
```

adamloup-enquizit reviewed (approved):
👍 Cool, can't wait to use it
